### PR TITLE
Commercial skipping notifications and toggle action

### PR DESCRIPTION
**Pull request by b-pass**
An implementation to addres this feature request from the forums: http://forum.kodi.tv/showthread.php?tid=208206

This adds a new action, TOGGLE_COMMSKIP which turns Commercial Skipping on/off for the current recording only (next record starts back at the default setting from settings.xml).

Also adds an option to display notifications of commercials and their lengths.

### CODE REVIEW

**Comment by b-pass** [comment]
Ok I did the localizations and fixed the whitespace.  I couldn't find any documentation on what localization numbers to use, so I just found some holes.  Is that ok?

**Comment by mkortstiege** [comment]
Please inverse the comskip option? It's easier to activate and disable instead of disable activation.

**Comment by FernetMenta** [comment]
support for comskip is no requirement for a pvr addon. hence adding those options to general settings is not correct. this needs to be enabled by something private to an addon and conveyed through the pvr api.

**Comment by ksooo** [comment]
I fully backup what @FernetMenta said. Currently we have a couple of general settings in the PVR area suffering from that problem. We "fix" this with the respective help text for the setting saying something like "not supported by all backends" or some other desperate words to workaround the actual design flaw. For new settings, we should do better.

File: xbmc/ApplicationPlayer.cpp [hunk 1]
```diff
@@ -257,6 +257,13 @@ bool CApplicationPlayer::SeekScene(bool bPlus)
   return (player && player->SeekScene(bPlus));
 }
 
+void CApplicationPlayer::ToggleCommSkip()
+{
+  boost::shared_ptr<IPlayer> player = GetInternal();
```
Comment (ksooo) [inline xbmc/ApplicationPlayer.cpp]: IMO, we consequently switched to std::shared_ptr some time ago?

**Comment by b-pass** [comment]
@ksooo, @FernetMenta CommSkip is supplied by Edit Decision Lists which can be supplied by an external (non-PVR) source as well.  PVR seems like the logical place for it.  I don't think this setting is addon-specific, but maybe it should go in videosettings instead?  Thoughts?

**Comment by ksooo** [comment]
The point is not about where to put the settings in the tree of Kodi general settings, but instead to put it into the settings of the addon actually supporting this feature.

The tricky part about this is, that the PVR addon API must be extended then with a function that allows Kodi to determine whether this option is available for the addon providing the current stream and that can be used by Kodi to obtain the value of the commskip setting.

Don't want to block this PR, just want to encourage people to do new things right, even if it is (much more) work. Please remember the 80/20 rule - sigh.

**Comment by b-pass** [comment]
@ksooo But this functionality is not only provided by the addon, it existed in XBMC/Kodi before PVR, and is supported also by flat files in several different formats.  Tying the ability to enable it to a PVR addon would break that existing functionality.

**Comment by MartijnKaijser** [comment]
in the case the commskip file is provided for normal files it's not even a PVR specific function so adding this setting in PVR section would already be pointless. 
I backup @FernetMenta and @ksooo that it should be brought through a PVR API. Yes it's more work but it the best way to do this

**Comment by Montellese** [comment]
I have no clue about the actual feature but from the description of @b-pass it sounds like this is just a toggle switch for people who already use this feature without any PVR addon installed simply using local files. So I don't understand why everyone is talking about extending the PVR API. Why would you want to extend the PVR API for a feature that can be used without having any PVR addon installed or PVR support completely disabled?

**Comment by b-pass** [comment]
@MartijnKaijser I moved the settings out of the PVR group and into the Video Library group to be near the other "TV" related settings.

File: system/settings/settings.xml [hunk 2]
```diff
@@ -463,6 +463,16 @@
           <default>false</default>
           <control type="toggle" />
         </setting>
+        <setting id="videoplayer.commskipenabled" type="boolean" label="38017" help="38018">
+          <level>1</level>
```
Comment (MartijnKaijser) [inline system/settings/settings.xml]: this in no way a lowest level setting. People who make EDL list are advanced users

**Comment by ksooo** [comment]
@Montellese Just talking about things to my best knowledge, dude. ;-) Did not know about the non-PVR aspects of the commskip feature, but know about the PVR functionality very well. That's why I'm talking about how to IMO do it right for the PVR part. 

The non-PVR provided EDLs makes things more complicated and I have no clue how to handle this part right, especially where to put any settings for this. Obviously, there is not always a "addon-specific" settings dialog available.

My points on how to do it IMO right for PVR-provided EDLs remain unchanged, though.

**Comment by MartijnKaijser** [comment]
The setting is not really clear that you need actual EDL files for commskip to work. This will only be a support burden because people think it will work out-of-the-box and come complaining.

**Comment by ksooo** [comment]
@MartijnKaijser yes, I'm actually afraid of this "support nightmare", too. We have to be really very clear about what this power user feature is/is not if we expose it through any UI.

**Comment by b-pass** [comment]
@ksooo @MartijnKaijser I moved the settings into advancedsettings.xml, there are a bunch of advanced commercial break settings for EDL already there so I think that ought to eliminate any support headaches.

**Comment by ksooo** [comment]
Much better, imo. Thx.

**Comment by mkortstiege** [comment]
Jep, looks better now.

**Comment by MartijnKaijser** [comment]
@xhaggi @Jalle19 @opdenkamp and who ever. Ok to go or last comments?

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Comment by opdenkamp** [comment]
needs a squash but looks okay. but this doesn't touch anything from pvr so you're pinging the wrong ppl ;-)

**Comment by b-pass** [comment]
Squashed.

**Comment by b-pass** [comment]
Squashed... I think?

**Comment by hudokkow** [comment]
This PR is a mess. You're using (your) master branch to work and you shouldn't.

Please create a new branch based on master, cherry-pick your commits and open a new pull request. We can close this one after that.

**Comment by b-pass** [comment]
Resubmitted as https://github.com/xbmc/xbmc/pull/7520

**Pull request closed without merging**